conbench/app/index.py

```diff
@@ -1,25 +1,76 @@
-import flask as f
+from collections import defaultdict
+from dataclasses import dataclass
+
+from sqlalchemy import select
 
 from ..app import rule
 from ..app._endpoint import AppEndpoint, authorize_or_terminate
 from ..app.benchmarks import RunMixin
 from ..config import Config
+from ..db import Session
+from ..entities.run import Run
 
 
 class Index(AppEndpoint, RunMixin):
-    def page(self, runs):
+    def page(self, repo_runs_map):
         return self.render_template(
             "index.html",
             application=Config.APPLICATION_NAME,
             title="Home",
-            runs=runs,
-            search_value=f.request.args.get("search"),
+            repo_runs_map=repo_runs_map,
+            # Note(JP): search_value is not consumed in template
+            # search_value=f.request.args.get("search"),
         )
 
     @authorize_or_terminate
     def get(self):
-        runs = self.get_display_runs()
-        return self.page(runs)
+        # Following
+        # https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#selecting-orm-entities
+        runs = Session.scalars(
+            select(Run).order_by(Run.timestamp.desc()).limit(1000)
+        ).all()
```

Comment:
`runs = self.get_display_runs()` is needless complexity, see API layer re-serialization.

This is for https://github.com/conbench/conbench/issues/890.

CC @austin3dickey 

btw using this SQLAlchemy 2.0 way of building up the query we get nice intellisense features, and proper type inference:
```
        runs = Session.scalars(
            select(Run).order_by(Run.timestamp.desc()).limit(1000)
        ).all()
```
![Screenshot from 2023-03-16 13-04-39](https://user-images.githubusercontent.com/265630/225916572-febc167b-1d92-4c30-ab07-71b70f63fdf7.png)

Reply:
> At the time I chose not to for a few reasons: it would take a little time ...

The right choice, let's do it gradually!